File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intv: List[List[int]]) -> int:
        intv.sort()
        start=intv[-1][0]
        ans=0
        for i in range(len(intv)-2, -1, -1):
            if intv[i][1] <= start:
                start = intv[i][0]
            else:
                ans+=1
        return ans
        

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intvl) {
        Arrays.sort(intvl, (a,b)->Integer.compare(a[0], b[0]));
        int l=intvl.length, start=intvl[l-1][0], remove=0;
        
        for(int i=l-2; i>=0; --i){
            if(start >= intvl[i][1]) start = intvl[i][0];
            else remove++;
        }
        
        return remove;
    }
}